### ISSUE TITLE
Symlink issue on k8s GPU node with /lib/firmware/nvidia/525.xx

### ISSUE THREAD

**Issue by maltegrosse** [GoogleContainerTools/kaniko#3006]
**Actual behavior**
After updating and upgrading to the latest nvidia drivers, kaniko runs into issues:
`error building image: error building stage: failed to get filesystem from image: error removing lib to make way for new symlink: unlinkat //lib/firmware/nvidia/525.147.05/gsp_ad10x.bin: device or resource busy`
- using --ignore-path=/lib/firmware/nvidia/525.147.05 is not considered, using /lib as ignore path breaks obvious other things during the build process

**Expected behavior**
A week ago, before the nvidia update, the container build run without issues.

**To Reproduce**
1. using k8s node with nvidia gpu driver installed
2. using [woodpecker-ci](https://woodpecker-ci.org/) with [kaniko plugin](https://github.com/woodpecker-ci/plugin-kaniko)

**Additional Information**
 - Dockerfile 

```
ARG LAB_IMAGE=quay.io/jupyter/scipy-notebook:lab-4.0.12
FROM ${LAB_IMAGE}
RUN pip install ipywebrtc==0.6.0 
```
 - Build Context
   No addition add/copy commands are used
By using ignote-path pointing to the exact firmware folder (/lib/firmware/nvidia/525.147.05), kaniko still shows DEBU[0001] Ignore list: .... {/lib/firmware/nvidia/525.147.05/gsp_ad10x.bin false}

 - Kaniko Image (fully qualified with digest)
gcr.io/kaniko-project/executor:v1.19.2-debug

https://github.com/woodpecker-ci/plugin-kaniko/blob/main/Dockerfile

 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [x] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [x] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [x] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [x] </li></ul>|

**Comment by marrotte** [comment]
Same problem, newer version:  //lib/firmware/nvidia/535.54.03/gsp_ga10x.bin

Coder envbuilder has a workaround for ignore-paths
https://github.com/coder/envbuilder/blob/8d3cfdffc3ab221a5d224418259128c46dd51a86/envbuilder.go#L544

But, after ignoring nvidia, I have to ignore /var/run and then this happens:

Failed to build: error building stage: failed to execute command: starting command: fork/exec /bin/sh: no such file or directory

**Comment by bvidovic1** [comment]
Same problem.

It works with docker build, but it does not work with kaniko.

**Comment by maltegrosse** [comment]
I couldnt get it running anymore, thats why I switched to buildah

**Comment by bvidovic1** [comment]
Yeah, I added additional CI templates based on DIND to build images with this problem (in my case nvidia/cuda) hoping it will get fixed soon.

**Comment by cvsik** [comment]
Are there any updates on this issue? I'd really like to stick with kaniko for my builds... 😢

**Comment by lahwaacz** [comment]
Seems related to https://github.com/GoogleContainerTools/kaniko/issues/1745

Adding `--ignore-path=/lib` works as a workaround :shrug:

**Comment by cvsik** [comment]
> Seems related to #1745
> 
> Adding `--ignore-path=/lib` works as a workaround 🤷

At least for me, this does not work as a workaround. Adding `--ignore-path=/lib` then gives me:
```
error building image: error building stage: failed to execute command: starting command: fork/exec /usr/local/bin/_dockerfile_shell.sh: no such file or directory
```

This was also mentioned in the comments above already 🤷

**Comment by lahwaacz** [comment]
@cvsik Oh, you're right. I thought the subsequent `failed to execute command` was another issue...

**Comment by Borg93** [comment]
any progress with this?
